pkg/v1/mutate/mutate.go

```diff
@@ -128,22 +133,27 @@ func Append(base v1.Image, adds ...Addendum) (v1.Image, error) {
 
 // Config mutates the provided v1.Image to have the provided v1.Config
 func Config(base v1.Image, cfg v1.Config) (v1.Image, error) {
-	m, err := base.Manifest()
+	cf, err := base.ConfigFile()
 	if err != nil {
 		return nil, err
 	}
 
-	cf, err := base.ConfigFile()
+	cf.Config = cfg
+
+	return ConfigFile(base, cf)
+}
+
+// ConfigFile mutates the provided v1.Image to have the provided v1.ConfigFile
+func ConfigFile(base v1.Image, cfg *v1.ConfigFile) (v1.Image, error) {
```

Comment:
We used Config in place of ConfigFile because it contained information the user might want to tweak, and getting it wrong made it a lot harder to screwup the image (e.g. it has diff ids).

Can you explain how you expect this to be (safely) used?  If it's for use within this package, let's make it private.

Reply:
making it private is probably the safest move, especially since I added the Canonical wrapper, which is what kaniko is going to call.